### ISSUE TITLE
iOS: Grid view

### ISSUE THREAD

**Issue by samuelclay** [samuelclay/NewsBlur#1720]
I think it'd be great to offer the grid view as an option on the story titles list. Re-use the same icons as well between List and Grid.

The grid view will also needs a stories per row option, probably between 2-4 and set to auto.

**Comment by Dejal** [comment]
This is a work in progress in the Catalyst branch; I'll merge into the Dejal branch once finished. So you can ignore it for now.

**Comment by Dejal** [comment]
Still a work in progress; it'll take a while. The feed detail now uses a modern diffable data source collection view, with a list layout for the old left/top/bottom layouts, and a custom layout for the new grid layout. Still lots of bugs, and the cells need work, but it's making good progress.

As a bonus, once this is done, we will be able to add the magazine and other layouts quite easily.

**Comment by Dejal** [comment]
Getting closer; almost usable now, though still some layout stuff to do, and issues to fix.

**Comment by Dejal** [comment]
The `UICollectionView`-based approach was coming together well enough, but had significant performance issues, layout problems, and jankiness, especially when scrolling — it'd jump around and be generally unacceptable.

So I decided to experiment with another approach. The Catalyst branch is all about dragging the app (kicking and screaming) into the modern era, so here's one more step along that path: I have rewritten the list and grid views in **SwiftUI**.

This experiment has gone extremely well — in way less code, the list and grid views perform superbly. There are still some features to reimplement, and behaviors to tweak (e.g. currently ignoring prefs for themes, preview position, etc), but it's already better than the previous collection view approach (and the old table view, which couldn't support the grid view).

I know I've been working on the grid view for a while, between urgent issues, but I think it's coming together nicely now.

![grid](https://user-images.githubusercontent.com/1318849/213889168-8dc2ea1c-47d1-455b-b5f2-c95dbea54343.png)

![list](https://user-images.githubusercontent.com/1318849/213889141-e32693d1-e3dd-4589-b5ae-2681f5c0d925.png)

**Comment by samuelclay** [comment]
Wow, this is amazing! I'd like to try something different, would you be able to list out all of the known issues / UI to develop as a checklist in this ticket? That way I can simply look at the single checklist (as part of the top ticket description) and not comment on anything that it in the works? I only want to give useful feedback but it's clear you know of plenty of things, so for such a big ticket item, let's try listing things out. We can always add to that checklist.

**Comment by Dejal** [comment]
Thanks. Yes, I keep a list of development notes in OmniFocus; fairly general at this experimental stage. No point in commenting on the current state, but I'll write up a list of remaining issues the next time I work on it.

**Comment by Dejal** [comment]
Lots more improvements (see commit message for a summary).

![Simulator Screen Shot - iPad Pro (11-inch) (4th generation) - 2023-02-02 at 20 59 50](https://user-images.githubusercontent.com/1318849/216505297-ab5870ea-a8ea-4758-a0bd-3d68c9c279a6.png)

![Simulator Screen Shot - iPad Pro (11-inch) (4th generation) - 2023-02-02 at 20 03 40](https://user-images.githubusercontent.com/1318849/216505224-987211d4-850e-426f-8752-6c0a6087a60e.png)

Here is a list of the current known issues. I may have forgotten some. No need to report anything yet, but if you do play with it, feel free to mention things to add to the list:

### General

- Implement mark read on scroll and selection.
- Reduce the reloads; tapping on a row reloads several times, unnecessarily (in older code).
- Remove the collection view stuff.
- Tidy up the code.
- Compare to the web, and fix any discrepancies.
- Check on iPhone; need to check List layout, but does Grid layout even make sense on iPhone?

### List layout

- Implement compact / comfortable settings.
- Implement content length settings.
- Swipe actions doesn’t work in LazyVGrid; instead could use a third-party solution. Though swipe actions don’t make sense for Grid layout, so maybe should change to a context menu for both?
- Scrolling the story view should scroll the list to show the selected story.

### Grid layout

- Launching the app in List layout then switching to Grid layout will sometimes not show the story content.

### Card view

- Implement themes.
- Implement font size settings.
- Implement context menu.

### Story view

- Adjust the height of the story view in grid layout based on the height of the story.
- Toggling a story between Text and Story views should resize the story cell.
- Scrolling between story pages should resize the story cell.

**Comment by Dejal** [comment]
More progress. Here are the remaining known issues:

Here is a list of the current known issues. I may have forgotten some. No need to report anything yet, but if you do play with it, feel free to mention things to add to the list:

### General

- Implement mark read on scroll and selection.
- Reduce the reloads; tapping on a row reloads several times, unnecessarily (in older code).
- Remove the collection view stuff.
- Tidy up the code.
- Compare to the web, and fix any discrepancies.
- Check on iPhone; need to check List layout, but does Grid layout even make sense on iPhone?

### List layout

- Swipe actions doesn’t work in LazyVGrid; instead could use a third-party solution. Though swipe actions don’t make sense for Grid layout, so maybe should change to a context menu for both?
- Scrolling the story view should scroll the list to show the selected story.

### Grid layout

- Launching the app in List layout then switching to Grid layout will sometimes not show the story content.

### Card view

- Implement context menu.

### Story view

- Not consistently scrolling to the story on selection.

**Comment by samuelclay** [comment]
Gorgeous! Some short notes:
* Bump up the feed title and date/author size by a point
* I like the loss of borders between story titles
* Let's also put in a modern slide-to-read/save
* Since we're now using swiftui, can we get animations between compact/comfortable states like we have on the web?

**Comment by Dejal** [comment]
For the "slide-to-read/save", do you mean a swipe gesture? I don't think that makes sense for the Grid layout, and it isn't natively supported in the SwiftUI view, so I was thinking it might be better to switch to a context menu instead. I have found some third-party code to implement a swipe gesture, if you really want to keep it on the List layout, but this could be a good opportunity to use a modern context menu. Let me know.

**Comment by Dejal** [comment]
More progress. All features have been implemented, just some cleanup and fixes remaining:

### Known issues

- Remove the collection view stuff.
- Tidy up the code.
- Launching the app in List layout then switching to Grid layout will sometimes not show the story content. (Spent some time investigating, but haven't solved it yet.)

Let me know what you think of the context menu as a possible replacement for swipes, given my previous comments.

**Comment by samuelclay** [comment]
There's a slight pause when scrolling down and marking as read. Also, let's make the line where a story is marked as read as half-way down the title and not at the very top edge (and not the bottom edge either). So you can see it mark as read, but it won't be distracting as you're reading and scrolling.

Also, the feed title and dates look the same size, even though I see in the commit history that they should be bigger.

Lastly, I like everything in the context menu and I think we should keep it, but I miss swipes and want to use the same style Apple is using in Mail, where a long swipe does the action.

**Comment by samuelclay** [comment]
Can I turn on the grid view on iPhone?

**Comment by Dejal** [comment]
Grid view isn't available on iPhone, as there isn't really room. It should be feasible to make it available, but I can't imagine it'd work well. Maybe if limited to two cards across, and only included a small image and title. I can try it if you wish.

I'll adjust the mark-as-read.

The feed title and dates are one point larger; I could bump it up another point.

And I'll add the third-party swipe solution.

**Comment by samuelclay** [comment]
I was thinking 1 card across, so it's really closer to magazine view than grid view.

**Comment by samuelclay** [comment]
But yeah, let's try both and make it configurable to see if it's workable

**Comment by Dejal** [comment]
Done those changes. I think Grid layout on iPhone works okay after all. Finding a third-party swipe that worked was a challenge.

All features have been implemented, just some cleanup and fixes remaining:

### Known issues

- Finish tiding up the code (the collection view stuff commented out for now, in case any of it is needed).
- Launching the app in List layout then switching to Grid layout will sometimes not show the story content. (Spent a little more time investigating, but haven't solved it yet; that's my next priority.)

**Comment by samuelclay** [comment]
Looking great! Some notes and crashes:

* Glad to have gestures again! These engage a fair bit less reliably than the current app. As I drag a story title, it moves slower than my finger, and then the gesture gets stuck when i drag my fingers downwards and lift up in order to cancel. I can attach a video if you like. Mainly I want to emulate the current app's gesture physics
* I'm getting a crash while reading a story, the `hash` is nil, so this line crashes in StoriesCollection.m:
    ```
    NSString *hash = story[@"story_hash"];
    NSString *title = story[@"story_title"];
    
    if (self.recentlyReadHashes[hash]) {
        NSLog(@"🔧 trying to sync as read when already read: %@: %@", hash, title);  // log
        return;
    }
    self.recentlyReadHashes[hash] = [NSString stringWithFormat:@"IN PROGRESS - %@", title];
    ```
* Crash when I hit "Send this story to..." when long pressing on a story title
* Can we add the pulsing blue indicator back in while loading stories?
* Similarly, where is the end story titles bottom indicator?
* I'd like to smoothly fade between unread and read
* The story content preview is too small M+L+XL. I think the size in XL should be in M, and then L and XL should each be 1-2pt successively larger
* Tapping on the story title list when the top right settings menu is open used to not open the story but would instead only close the popover. Now it does both. Tapping off the settings popover should not activate what's behind it. The feed list view does this correctly.
* When returning back to the story titles list from story detail, the wrong story is selected and there is an animation flash
* Once when returning to the list I got this screen:
    ![IMG_3889](https://user-images.githubusercontent.com/44229/230790278-080052a7-72fa-4869-b1f0-4504a1198fe3.PNG)
* Unread icon is too large. Compared to the web it's about 1.5X sized:
    ![IMG_3887](https://user-images.githubusercontent.com/44229/230790323-a582dc27-798d-4aeb-9315-92ff64597447.PNG)
    <img width="1049" alt="Screenshot 2023-04-09 at 2 30 58 PM" src="https://user-images.githubusercontent.com/44229/230790337-5b8546c1-2f68-4033-b761-3a2290f80e45.png">
* In trying to take screenshots between both the current app and the catalyst app, I seemed to have bricked the app and no longer will All Site Stories load, I just see this:
    ![IMG_3890](https://user-images.githubusercontent.com/44229/230790546-694e9c65-6b90-4f72-aeb9-0aea8bc4b0c8.PNG)
* When using edge swipe on the story titles back to the feed titles, the Read Story gesture engages. I included a video, which also includes the issue of the gesture staying engaged even after I let go.

https://user-images.githubusercontent.com/44229/230790580-1708b849-90bb-46ec-9b1d-945d01104736.MP4

**Comment by Dejal** [comment]
For some reason I didn't see your comment until just now, so I didn't address some of those issues. I'll get to them next time. Other than those, there are no more known issues. Feel free to do more testing and add more feedback.

**Comment by samuelclay** [comment]
Loading it up today but it crashed on boot on my iPad Air:

```
preferredStatusBarStyle: light
preferredStatusBarStyle: light
preferredStatusBarStyle: light
preferredStatusBarStyle: light
preferredStatusBarStyle: light
2023-04-30 19:53:41.709810-0400 NewsBlur Dev[3300:2259916]  ---> DB dir: file:///var/mobile/Containers/Data/Application/A9EA3B83-D416-4767-86FB-08C1E8F304C5/Documents/
2023-04-30 19:53:41.710569-0400 NewsBlur Dev[3300:2259916] Dropped db: not an error
2023-04-30 19:53:41.763068-0400 NewsBlur Dev[3300:2260331] Task <9B1751FA-2138-4D6D-A5C2-ED001C167019>.<1> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=(null), NSErrorFailingURLKey=(null), _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <9B1751FA-2138-4D6D-A5C2-ED001C167019>.<1>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <9B1751FA-2138-4D6D-A5C2-ED001C167019>.<1>, NSUnderlyingError=0x283acc060 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2023-04-30 19:53:41.776675-0400 NewsBlur Dev[3300:2259916] Loading offline feeds: 0
2023-04-30 19:53:41.782003-0400 NewsBlur Dev[3300:2260331] Reachable: <Reachability: 0x28212cb40 (-R -t-----)>
preferredStatusBarStyle: light
preferredStatusBarStyle: light
preferredStatusBarStyle: light
preferredStatusBarStyle: light
2023-04-30 19:53:41.945819-0400 NewsBlur Dev[3300:2259916] Metal API Validation Enabled
2023-04-30 19:53:42.086548-0400 NewsBlur Dev[3300:2260334] fopen failed for data file: errno = 2 (No such file or directory)
2023-04-30 19:53:42.086582-0400 NewsBlur Dev[3300:2260334] Errors found! Invalidating cache...
2023-04-30 19:53:42.106875-0400 NewsBlur Dev[3300:2260334] fopen failed for data file: errno = 2 (No such file or directory)
2023-04-30 19:53:42.106908-0400 NewsBlur Dev[3300:2260334] Errors found! Invalidating cache...
2023-04-30 19:53:42.120290-0400 NewsBlur Dev[3300:2259916] Loading offline feeds: 0
preferredStatusBarStyle: light
preferredStatusBarStyle: light
2023-04-30 19:53:42.165125-0400 NewsBlur Dev[3300:2259916] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x281714870 V:[UIView:0x102d424f0]-(-13)-|   (active, names: '|':UIView:0x102d42c80 )>",
    "<NSLayoutConstraint:0x281714d20 V:[UIView:0x102d07710]-(0)-|   (active, names: '|':UIView:0x102d42c80 )>",
    "<NSLayoutConstraint:0x281714d70 V:[UIView:0x102d424f0]-(0)-[UIView:0x102d07710]   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x281714d70 V:[UIView:0x102d424f0]-(0)-[UIView:0x102d07710]   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.
2023-04-30 19:53:42.216053-0400 NewsBlur Dev[3300:2259916] <SKPaymentQueue: 0x283650350>: No observers found that respond to "paymentQueue:shouldAddStorePayment:forProduct:", will not check for purchase intents
2023-04-30 19:53:42.217486-0400 NewsBlur Dev[3300:2260332] Task <3AC67F47-8117-44C8-B642-9D177459FDD4>.<2> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=(null), NSErrorFailingURLKey=(null), _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <3AC67F47-8117-44C8-B642-9D177459FDD4>.<2>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <3AC67F47-8117-44C8-B642-9D177459FDD4>.<2>, NSUnderlyingError=0x283b65d10 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2023-04-30 19:53:42.218310-0400 NewsBlur Dev[3300:2259916] Fetching feed list
2023-04-30 19:53:42.230642-0400 NewsBlur Dev[3300:2259916] Reachability changed: NSConcreteNotification 0x2834e3460 {name = kReachabilityChangedNotification; object = <Reachability: 0x28212cb40 (-R -t-----)>}
2023-04-30 19:53:42.233851-0400 NewsBlur Dev[3300:2259916] Fetching feed list
2023-04-30 19:53:42.256210-0400 NewsBlur Dev[3300:2259916] Could not fetch user avatar: Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=(null), NSErrorFailingURLKey=(null), _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <3AC67F47-8117-44C8-B642-9D177459FDD4>.<2>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <3AC67F47-8117-44C8-B642-9D177459FDD4>.<2>, NSUnderlyingError=0x283b65d10 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
preferredStatusBarStyle: dark
preferredStatusBarStyle: dark
preferredStatusBarStyle: dark
2023-04-30 19:53:43.284986-0400 NewsBlur Dev[3300:2260336] Task <A27E86BF-8D90-4025-942B-68999C7CE7F0>.<3> finished with error [-1002] Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=(null), NSErrorFailingURLKey=(null), _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <A27E86BF-8D90-4025-942B-68999C7CE7F0>.<3>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <A27E86BF-8D90-4025-942B-68999C7CE7F0>.<3>, NSUnderlyingError=0x283ad21c0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
preferredStatusBarStyle: dark
preferredStatusBarStyle: dark
2023-04-30 19:53:43.299458-0400 NewsBlur Dev[3300:2259916] System changed to theme: Light
2023-04-30 19:53:43.302877-0400 NewsBlur Dev[3300:2259916] ⚠️ show column for loadRiverFeedDetailView: split view controller: <NewsBlur_Dev.SplitViewController: 0x102e0d740> split nav: (null); split controllers: (
    "<UINavigationController: 0x106809600>",
    "<UINavigationController: 0x103820a00>",
    "<UINavigationController: 0x103870a00>"
); detail controller: <NewsBlur_Dev.DetailViewController: 0x102d14360>; detail nav: <UINavigationController: 0x103870a00>; detail nav controllers: (
    "<NewsBlur_Dev.DetailViewController: 0x102d14360>"
)
2023-04-30 19:53:43.302945-0400 NewsBlur Dev[3300:2259916] ...shown
2023-04-30 19:53:43.324795-0400 NewsBlur Dev[3300:2259916] Could not fetch user avatar: Error Domain=NSURLErrorDomain Code=-1002 "unsupported URL" UserInfo={NSLocalizedDescription=unsupported URL, NSErrorFailingURLStringKey=(null), NSErrorFailingURLKey=(null), _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <A27E86BF-8D90-4025-942B-68999C7CE7F0>.<3>"
), _NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <A27E86BF-8D90-4025-942B-68999C7CE7F0>.<3>, NSUnderlyingError=0x283ad21c0 {Error Domain=kCFErrorDomainCFNetwork Code=-1002 "(null)"}}
2023-04-30 19:53:44.166205-0400 NewsBlur Dev[3300:2260336] Saving scroll 1 in (null)-(null)
2023-04-30 19:53:45.705055-0400 NewsBlur Dev[3300:2259916] [User Defaults] Couldn't read values in CFPrefsPlistSource<0x281a2eb80> (Domain: group.com.newsblur.NewsBlur-Group, User: kCFPreferencesAnyUser, ByHost: Yes, Container: (null), Contents Need Refresh: Yes): Using kCFPreferencesAnyUser with a container is only allowed for System Containers, detaching from cfprefsd
2023-04-30 19:53:45.883754-0400 NewsBlur Dev[3300:2260330] *** Terminating app due to uncaught exception 'NSGenericException', reason: '*** Collection <__NSArrayM: 0x283b87b70> was mutated while being enumerated.'
*** First throw call stack:
(0x1afd82d94 0x1a8eb43d0 0x1afeef9bc 0x1afd90924 0x1020f78b0 0x102b78520 0x102b7a038 0x102b8e670 0x102b8ee34 0x20ec4eda0 0x20ec4eb7c)
libc++abi: terminating due to uncaught exception of type NSException
```

And on my iPhone SE 3, tapping on story titles does nothing.

**Comment by samuelclay** [comment]
Oh man, swiping is so janky, can we use the built that other apps are using? Overcast and Mail.app both have identical swipes, let's use that!

**Comment by Dejal** [comment]
Unfortunately the built-in swipe action is not supported with the SwiftUI `LazyVGrid` currently. Hopefully they'll add it in the next iOS update; in the meantime we need to use a third-party solution. I tried several, and they were all kinda dodgy in different ways; I could look some more, or try to improve it, or we could live with it and hope Apple provides support soon.

**Comment by samuelclay** [comment]
We made a mistake then because we can't launch without swipe gestures and janky gestures are not happening. I think we should table this branch until grid's have native swipe gestures.

**Comment by Dejal** [comment]
See the commit summaries for those three commits; the old table view has been restored for the list layout, and a bunch of things have been fixed and improved.

Subject to further tweaks or fixes, this feature is complete.

**Comment by Dejal** [comment]
More fixes and tweaks; see commit details.

Re an interactive swipe across screen to return to the feeds list — that is already available as an edge swipe. Which is distinct from a non-edge swipe, which changes story pages.

The slight hesitation when scrolling is due to reloading the view when marking as read on scroll. The grid view now supports disabling that, which is faster. It's a complex view, so I'm not sure I could improve that.

I tried to implement a fade on unread to read, but couldn't get it to work well yet.

**Comment by samuelclay** [comment]
Let's do something about the performance of scrolling because it needs to be smoother. If that means queueing up marking as read and then committing it to the db later, then that's fine.

I'd like to have a non-edge interactive swipe back to the feed list while in Grid view. That's the tradeoff for not having gestures. Other readers have it and it leads to a more premium feel.

**Comment by Dejal** [comment]
I have some ideas on increasing scrolling performance. It may require deferring marking as read as you say, though there are some other possibilities. It'll take some time to work through optimizing it. I'll work on that next time.

I've created a separate issue for the new feature request.

**Comment by samuelclay** [comment]
Opening a story, advancing a couple stories, and then returning to the list view doesn't show the correct story. 

https://github.com/samuelclay/NewsBlur/assets/44229/31cd9cfe-6f8b-49b5-9df8-6d6f001339cb

**Comment by samuelclay** [comment]
Also getting a crash on load

https://github.com/samuelclay/NewsBlur/assets/44229/3e53005e-a22d-4d32-aac4-80aecf1dd000

**Comment by samuelclay** [comment]
Amazing, I can really feel the difference. And those bug fixes are well appreciated! Two changes left before I feel this is ready for TestFlight:

1. The grey theme needs a grid story background color that is different from the table background. It should be probably be a lighter color than the background, so darken the table background in the grey theme.
2. The grid vs list view should be set per-folder/per-feed, like on the web. Some folders are better in List view, some in Grid view.

**Comment by samuelclay** [comment]
Another issue I just discovered shows that some stories are incorrectly being marked as read. See this video:


https://github.com/samuelclay/NewsBlur/assets/44229/61a3233c-71cf-4a66-8614-25b50ee50854

**Comment by samuelclay** [comment]
Should I be posting these videos as separate new issues?

Here's another issue, the story title highlight is jarringly deselected and slew eyed when I swipe back from the story detail view. I'd like it to show as highlighted and not fade it's highlight until the story titles list is fully in view. 

https://github.com/samuelclay/NewsBlur/assets/44229/835260a6-c89a-44e2-ab80-ea389b98823a

**Comment by Dejal** [comment]
Including videos in this issue is fine, if related to the grid view. Those two are with the list variation, but close enough.

**Comment by samuelclay** [comment]
I'm also getting a near constant crash when scrolling, even sometimes on the first page! Here's the debug output, thought it took a number of pages to trigger this one:

```
Loaded story 'The House G.O.P. Bomb-throwers May End Up Bombing Their Majority
Loaded story 'The House G.O.P. Bomb-throwers May End Up Bombing Their Majority
Loaded story 'Democrats to Press Supreme Court Ethics Rules Over G.O.P. Opposition
Loaded story 'Democrats to Press Supreme Court Ethics Rules Over G.O.P. Opposition
Loaded story 'Kai Bird on Christopher Nolan's ‘Oppenheimer’
Loaded story 'Kai Bird on Christopher Nolan's ‘Oppenheimer’
Loaded story 'Hubert Humphrey’s Democratic Convention Speech Resonates Today
Loaded story 'Hubert Humphrey’s Democratic Convention Speech Resonates Today
Loaded story 'Trump’s Judges: More Religious Ties and More N.R.A. Memberships
Loaded story 'Trump’s Judges: More Religious Ties and More N.R.A. Memberships
Loaded story 'Mural by Safe in Moyobamba, Peru for TierraQPinta
Loaded story 'Mural by Safe in Moyobamba, Peru for TierraQPinta
Loaded story 'Brussels-based telecom operator Proximus Group plans to acquire 58% stake in Indian cloud communication service provider Route Mobile for $721M (Alex Gabriel Simon/Bloomberg)
Loaded story 'Brussels-based telecom operator Proximus Group plans to acquire 58% stake in Indian cloud communication service provider Route Mobile for $721M (Alex Gabriel Simon/Bloomberg)
Loaded story 'Kerry Says U.S. and China Must Set Aside Politics to Tackle Climate Change
Loaded story 'Kerry Says U.S. and China Must Set Aside Politics to Tackle Climate Change
Loaded story 'SwiftUI onChange Deprecation
Loaded story 'SwiftUI onChange Deprecation
Loaded story 'A jail escapee evaded authorities for days. Then a dog started barking.
Loaded story 'A jail escapee evaded authorities for days. Then a dog started barking.
Loaded story 'Tokyo is the new Paris
Loaded story 'Tokyo is the new Paris
Loaded story 'Samsung Debuts New $1,599 ViewFinity S9 5K Display to Compete with Apple's Studio Display
Loaded story 'Samsung Debuts New $1,599 ViewFinity S9 5K Display to Compete with Apple's Studio Display
Loaded story 'Samsung prices its 27-inch 5K monitor at $1,599 — just like Apple’s Studio Display
Loaded story 'Samsung prices its 27-inch 5K monitor at $1,599 — just like Apple’s Studio Display
Loaded story 'DIY Laser For Ablating Metal
Loaded story 'DIY Laser For Ablating Metal
Loaded story 'Murder conviction reversed over relationship between judge and prosecutor
Loaded story 'Murder conviction reversed over relationship between judge and prosecutor
Loaded story 'Selling My Drawings
Loaded story 'Selling My Drawings
Loaded story 'Webb may have spotted "dark stars" made of annihilating dark matter
Loaded story 'Webb may have spotted "dark stars" made of annihilating dark matter
Loaded story 'Exercise during dialysis treatment delivers big health benefits
Loaded story 'Exercise during dialysis treatment delivers big health benefits
Loaded story 'Exchange Online Mailbox Enablement via Ansible
Loaded story 'Exchange Online Mailbox Enablement via Ansible
Loaded story 'NBA (CEO) fact of the day
Loaded story 'NBA (CEO) fact of the day
Loaded story 'Saudi Arabia’s Neom Seeks $2.7 Billion Loan to Build Futuristic City
Loaded story 'Saudi Arabia’s Neom Seeks $2.7 Billion Loan to Build Futuristic City
Loaded story 'Women’s World Cup Contenders
Loaded story 'Women’s World Cup Contenders
pref change for 'NBA (CEO) fact of the day': [NewsBlur_Dev.CardFrame(id: "55413834-5F68-44A2-83EE-FBFA43AFB2FC", frame: (0.0, 533.5, 375.0, 149.0))]
pref change for 'Exercise during dialysis treatment delivers big health benefits': [NewsBlur_Dev.CardFrame(id: "ED7EE790-BF71-4125-9DCC-183FC8E8E2D2", frame: (0.0, 524.5, 375.0, 167.0))]
pref change for 'Exchange Online Mailbox Enablement via Ansible': [NewsBlur_Dev.CardFrame(id: "049C615C-9088-4C8B-A63E-D2E46C00BCE4", frame: (0.0, 524.5, 375.0, 167.0))]
pref change for 'Saudi Arabia’s Neom Seeks $2.7 Billion Loan to Build Futuristic City': [NewsBlur_Dev.CardFrame(id: "AEEA6FC1-582D-4C41-BC1E-8EF07A46C0F9", frame: (0.0, 524.5, 375.0, 167.0))]
pref change for 'Women’s World Cup Contenders': [NewsBlur_Dev.CardFrame(id: "49966826-9986-4B32-AA39-8C45343F2AA4", frame: (0.0, 575.0, 375.0, 134.0))]
pref change for 'NBA (CEO) fact of the day': [NewsBlur_Dev.CardFrame(id: "55413834-5F68-44A2-83EE-FBFA43AFB2FC", frame: (0.0, 249.5, 375.0, 148.5))]
pref change for 'Exercise during dialysis treatment delivers big health benefits': [NewsBlur_Dev.CardFrame(id: "ED7EE790-BF71-4125-9DCC-183FC8E8E2D2", frame: (0.0, -84.0, 375.0, 166.5))]
pref 'Exercise during dialysis treatment delivers big health benefits': scrolled off the top
pref change for 'Exchange Online Mailbox Enablement via Ansible': [NewsBlur_Dev.CardFrame(id: "049C615C-9088-4C8B-A63E-D2E46C00BCE4", frame: (0.0, 82.5, 375.0, 167.0))]
pref change for 'Saudi Arabia’s Neom Seeks $2.7 Billion Loan to Build Futuristic City': [NewsBlur_Dev.CardFrame(id: "AEEA6FC1-582D-4C41-BC1E-8EF07A46C0F9", frame: (0.0, 398.0, 375.0, 167.0))]
pref change for 'Women’s World Cup Contenders': [NewsBlur_Dev.CardFrame(id: "49966826-9986-4B32-AA39-8C45343F2AA4", frame: (0.0, 565.0, 375.0, 134.0))]
NBA (CEO) fact of the day appeared
Exercise during dialysis treatment delivers big health benefits appeared
Exchange Online Mailbox Enablement via Ansible appeared
Saudi Arabia’s Neom Seeks $2.7 Billion Loan to Build Futuristic City appeared
Women’s World Cup Contenders appeared
2023-07-17 07:21:25.440719-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b36b0f6800> {length = 2, path = 0 - 70}
2023-07-17 07:21:27.954977-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b36a0f6800> {length = 2, path = 0 - 71}
2023-07-17 07:21:28.339774-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3650f6800> {length = 2, path = 0 - 72}
2023-07-17 07:21:31.341176-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3640f6800> {length = 2, path = 0 - 73}
2023-07-17 07:21:31.605562-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3670f6800> {length = 2, path = 0 - 74}
2023-07-17 07:21:32.374365-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3660f6800> {length = 2, path = 0 - 75}
2023-07-17 07:21:33.675338-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3610f6800> {length = 2, path = 0 - 76}
2023-07-17 07:21:33.805668-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3600f6800> {length = 2, path = 0 - 77}
2023-07-17 07:21:33.879706-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C31.1.1:3] flags=[R.] seq=336352211, ack=2562103176, win=133 state=LAST_ACK rcv_nxt=336352211, snd_una=2562103176
2023-07-17 07:21:33.880053-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C31.1.1:3] flags=[R] seq=336352211, ack=0, win=0 state=CLOSED rcv_nxt=336352211, snd_una=2562103176
2023-07-17 07:21:33.880165-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C31.1.1:3] flags=[R] seq=336352211, ack=0, win=0 state=CLOSED rcv_nxt=336352211, snd_una=2562103176
2023-07-17 07:21:33.953445-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C45.1.1:3] flags=[R.] seq=2454718442, ack=3196752239, win=133 state=LAST_ACK rcv_nxt=2454718442, snd_una=3196752239
2023-07-17 07:21:33.953911-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C45.1.1:3] flags=[R] seq=2454718442, ack=0, win=0 state=CLOSED rcv_nxt=2454718442, snd_una=3196752239
2023-07-17 07:21:33.988150-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3630f6800> {length = 2, path = 0 - 78}
2023-07-17 07:21:34.222658-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3620f6800> {length = 2, path = 0 - 79}
2023-07-17 07:21:34.792099-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C508.1.1:3] flags=[R] seq=1638345918, ack=0, win=0 state=LAST_ACK rcv_nxt=1638345918, snd_una=2895122508
2023-07-17 07:21:34.792501-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C508.1.1:3] flags=[R] seq=1638345918, ack=0, win=0 state=CLOSED rcv_nxt=1638345918, snd_una=2895122508
2023-07-17 07:21:34.809563-0400 NewsBlur Dev[52206:3100072] [tcp] tcp_input [C508.1.1:3] flags=[R] seq=1638345918, ack=0, win=0 state=CLOSED rcv_nxt=1638345918, snd_una=2895122508
2023-07-17 07:21:34.923186-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37d0f6800> {length = 2, path = 0 - 80}
2023-07-17 07:21:35.172623-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37c0f6800> {length = 2, path = 0 - 81}
2023-07-17 07:21:35.237902-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37f0f6800> {length = 2, path = 0 - 82}
2023-07-17 07:21:35.370380-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37e0f6800> {length = 2, path = 0 - 83}
2023-07-17 07:21:35.554763-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3790f6800> {length = 2, path = 0 - 84}
2023-07-17 07:21:35.889765-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3780f6800> {length = 2, path = 0 - 85}
2023-07-17 07:21:35.937669-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37b0f6800> {length = 2, path = 0 - 86}
2023-07-17 07:21:35.981266-0400 NewsBlur Dev[52206:3100072] [connection] nw_connection_add_timestamp_locked_on_nw_queue [C531] Hit maximum timestamp count, will start dropping events
2023-07-17 07:21:36.020600-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b37a0f6800> {length = 2, path = 0 - 87}
2023-07-17 07:21:36.085922-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3750f6800> {length = 2, path = 0 - 88}
2023-07-17 07:21:36.162989-0400 NewsBlur Dev[52206:3097869] finishedLoadingFeed: (null)
2023-07-17 07:21:36.163589-0400 NewsBlur Dev[52206:3097869] ...rendered
2023-07-17 07:21:36.163655-0400 NewsBlur Dev[52206:3097869] ...advanced to next unread
2023-07-17 07:21:36.204596-0400 NewsBlur Dev[52206:3097869]  --> Reloading indexPath: <NSIndexPath: 0x881b35b3740f6800> {length = 2, path = 0 - 89}
2023-07-17 07:21:36.214993-0400 NewsBlur Dev[52206:3097869] *** Assertion failure in -[UITableView _Bug_Detected_In_Client_Of_UITableView_Invalid_Number_Of_Rows_In_Section:], UITableView.m:2620
2023-07-17 07:21:36.217975-0400 NewsBlur Dev[52206:3097869] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid update: invalid number of rows in section 0. The number of rows contained in an existing section after the update (105) must be equal to the number of rows contained in that section before the update (94), plus or minus the number of rows inserted or deleted from that section (1 inserted, 1 deleted) and plus or minus the number of rows moved into or out of that section (0 moved in, 0 moved out). Table view: <UITableView: 0x109825000; frame = (0 0; 375 603); clipsToBounds = YES; autoresize = RM+BM; gestureRecognizers = <NSArray: 0x283184960>; backgroundColor = UIExtendedSRGBColorSpace 0.0431373 0.0431373 0.0431373 1; layer = <CALayer: 0x283f239a0>; contentOffset: {0, 14945.5}; contentSize: {375, 15644}; adjustedContentInset: {0, 0, 0, 0}; dataSource: <NewsBlur_Dev.FeedDetailViewController: 0x105879c00>>'
```

**Comment by samuelclay** [comment]
After your latest commit, the crashes have been resolved, but the stories being incorrectly marked as read is still occuring: https://github.com/samuelclay/NewsBlur/issues/1720#issuecomment-1622535279. 

Also, the App Store connect build is still failing with an Invalid Binary warning. Are you getting these warnings? If not, let me add you to the team on the App Store so you can resolve it. I'd love to be getting NewsBlur Dev automatically over test flight.

**Comment by Dejal** [comment]
I wasn't able to reproduce the stories incorrectly marked as read. I'll try again, but if you have any clues, let me know.

I get the NewsBlur Dev Xcode Cloud error too. I had a play to see if I could figure out the issue, but wasn't successful. It seems to have gotten confused due to the production and Dev editions, complaining that the bundle identifier can't be changed, even though it is correct. We may need to change it again and add a new listing in App Store Connect to resolve it. I don't have access to do that, though.

**Comment by Dejal** [comment]
I've spent some time on Xcode Cloud, and have gotten it to work. **NewsBlur Dev** is now **NewsBlur Alpha**, and will now automatically update in TestFlight for the two of us for internal testing. You will have received an invite to join the new TestFlight.

(You can delete the [**NewsBlur Dev**](https://appstoreconnect.apple.com/apps/1667569841/appstore/info) listing from App Store Connect if you wish.)

**Comment by samuelclay** [comment]
There's a number of issues with the Grid view on iPad:

* Dates are wrong
* Duplicate stories (see screenshot)
* Tapping on a story shows a different story in the title bar (see screenshot)
* No story visible in the story content
* Closing a story by tapping on the story title bar leaves you in a random place
* I got into a state where the "Feed list" button in the top left of the story titles list didn't do anything, and I couldn't swipe back because it would bring up the new Stage manager

![IMG_0022](https://github.com/samuelclay/NewsBlur/assets/44229/1b0e2094-79b4-4072-b48c-68a3d6cea274)
![IMG_0024](https://github.com/samuelclay/NewsBlur/assets/44229/8d340339-a769-4c0c-9809-90db72719260)

**Comment by samuelclay** [comment]
On the latest update, I'm seeing a theme jump from dark to light to dark. I'm on auto dark and it's 5am so it should be dark. 

https://github.com/samuelclay/NewsBlur/assets/44229/7930c89a-dcdd-4717-b2f9-020bf4c668a2

**Comment by samuelclay** [comment]
Here's another issue, I've found that the all site stories list&grid will sometimes show a loading bar and never actually load. It feels like the stories have come in due to the way it blinks rapidly three times before it goes into an infinite loop of a slow fade. Here's a video showing it repeatedly in action. 

https://github.com/samuelclay/NewsBlur/assets/44229/08137ae3-9289-4079-be91-298b5836d59a

**Comment by samuelclay** [comment]
Last issue I could find: persistent background crashes. I've been sharing them through TestFlight so hopefully we can get a stack trace through that.

**Comment by samuelclay** [comment]
A small issue is that story titles don't seamlessly transition between read states like they used to. This video shows the difference. 

https://github.com/samuelclay/NewsBlur/assets/44229/d3b780c8-7016-452c-b78e-125dfd151639

**Comment by Dejal** [comment]
I've addressed those issues. Give it a go and let me know if anything remains.

**Comment by samuelclay** [comment]
On iPhone using grid view, tapping on a story and immediately swiping back to the story list shows the wrong story. See this video. This is on yesterdays build. 

https://github.com/samuelclay/NewsBlur/assets/44229/0f64a206-b009-401c-b90f-e35d9863095f

**Comment by samuelclay** [comment]
Also, on iPad, there is a lot of glitchiness. Do I need to record a video? I assume you see it as the grid view on ipad is not a smooth experience yet. I love swiping the stories with story titles inline, but the view can't move and redraw.

**Comment by Dejal** [comment]
I'll see what I can do to improve it. I do see some glitchiness, but videos or descriptions of specific issues do help me, since sometimes you experience something I haven't seen. For example, I hadn't seen the theme changing issue until after you pointed it out and I worked to reproduce it.

**Comment by Dejal** [comment]
Alrighty, give this a go. Let me know if you want any further changes.

**Comment by samuelclay** [comment]
There's still some issues with the theme selection in auto-theme. Here's a video showing the theme bouncing back and forth in the morning when the theme has switched but the views haven't updated. 

https://github.com/samuelclay/NewsBlur/assets/44229/7c2a6270-4447-414b-aab1-b89eb52413a5

**Comment by samuelclay** [comment]
The clocks on all of the stories are wrong. It matches the local time. ![image](https://github.com/samuelclay/NewsBlur/assets/44229/650a93dc-48c8-4af4-a375-aa20cc18e124)

**Comment by Dejal** [comment]
Okay, I've fixed the date issue, and I think I've fixed the theme issue (I wasn't able to reproduce it, but I think I know what was happening; seems a recent iOS behavior change).

Please confirm the theme is behaving correctly now. If it isn't, please show me your Theme Preferences (I assume you have "Follow system appearance" switched on).

Let me know if you spot any other issues. Otherwise you can merge this into the Master branch. It'd be nice to get this major new feature released, and move on to other improvements.

**Comment by samuelclay** [comment]
Great, I'll check it out. Shouldn't there have been a TF build? I don't see anything since the end of August, 20 days ago.

**Comment by Dejal** [comment]
Looks like you need to [agree to the latest developer membership agreement](https://developer.apple.com/account/#/membership) before Xcode Cloud can produce a build.

There might also be [a separate distribution agreement in App Store Connect](https://appstoreconnect.apple.com/agreements/#/).

**Comment by samuelclay** [comment]
Ok, both have been agreed to.

**Comment by Dejal** [comment]
Thanks. I started a manual build; it is now available in TestFlight.

**Comment by samuelclay** [comment]
The iPad app is looking great, no issues except one. I'm getting a background crash. I uploaded it to TF.

**Comment by Dejal** [comment]
Hmm, that crash is tricky, as it is happening deep in iOS in `UIKitCore`, with no NewsBlur code in the backtrace, and no indication of the cause. I'll see if I can reproduce it, in case I can gather any more information about it. Please log any additional experiences of it.

**Comment by Dejal** [comment]
I've had the iPad app running on my iPad connected to Xcode for the past two days, mostly in the background, with occasional foreground use, but it hasn't crashed at all. Can you provide any more clues on how I can reproduce it? Have you had any more crashes? I only see one crash log from TestFlight, which as mentioned is deep in iOS, without any indication as to cause.

**Comment by samuelclay** [comment]
I'm trying to get it on TestFlight and soon the App Store, but I need to get the Simulator working. I'm hitting an error when I try to take iPhone 15 screenshots in the simulator. Can you get it working there?

**Comment by Dejal** [comment]
Yes, I can. What error are you getting?

![Screenshot 2023-10-02 at 15 40 06](https://github.com/samuelclay/NewsBlur/assets/1318849/831cb70c-00cd-45df-afad-bc86fc71c3fc)

**Comment by samuelclay** [comment]
Tried restarting Xcode and Simulator to no avail...

<img width="1909" alt="Screenshot 2023-10-02 at 6 23 38 PM" src="https://github.com/samuelclay/NewsBlur/assets/44229/ea715b7c-6b18-4439-83df-9506694f4920">

**Comment by Dejal** [comment]
Have you tried restarting your Mac? That often fixes wackiness. 

Failing that, you could go to Xcode's Window ▸ Devices and Simulators menu item, choose Simulators, and click the + to add a new simulator for the device.

**Comment by samuelclay** [comment]
Work on Catalyst support, I'd love to have a NewsBlur widget in macOS again

**Comment by Dejal** [comment]
Sounds good. It'd be exciting to have a Mac edition of the app.

I have merged the `Catalyst` branch into `dejal`, as release-ready (or a candidate) code. You should build the TestFlight from the `dejal` branch. 

When you're ready to do the general release, you can merge `dejal` into `master`.

I'll use the `Catalyst` branch for Catalyst work, appropriately enough. Don't build from this branch; I'll continue to do the NewsBlur Alpha in-progress builds from this branch.

**Comment by samuelclay** [comment]
Yeah, especially looking forward to notifications on the Mac. I find myself reading the Apple News app occasionally because notifications come through. We have that on the web, but I want the native app.

**Comment by samuelclay** [comment]
Amazingly, Apple got back to me about the Simulator not working with this:
> Engineering has provided the following information regarding this issue:
> 
> App extension placeholders array has multiple placeholders with the same bundleID of com.newsblur.NewsBlur.widget.

**Comment by Dejal** [comment]
Ah, that's probably a merge issue. I think we had the old widget in the Dejal branch, and the modern widget in the Catalyst branch. Though I thought the modern widget was in a release, so maybe not. Anyway, we should probably remove the old widget. Or we could change its bundle ID if you want to keep it around.

**Comment by samuelclay** [comment]
Let's keep the old widget, just change it's bundle ID.

**Comment by Dejal** [comment]
Actually the bundle ID was already distinct, but some Catalyst changes had snuck in to the Dejal branch. Try it now.

**Comment by samuelclay** [comment]
> Actually the bundle ID was already distinct, but some Catalyst changes had snuck in to the Dejal branch. Try it now.

Did this go out on TestFlight using Xcode Cloud?

**Comment by aronatkins** [comment]
Demonstration showing how feeds appear empty, but then populate after revisiting. iOS beta using Version 13.0 (147).

https://github.com/samuelclay/NewsBlur/assets/362187/7188eda3-6195-4e9d-b399-3e9f65905960

**Comment by samuelclay** [comment]
Thanks for posting that, it's incredibly helpful!

**Comment by Dejal** [comment]
I fixed some crashes, but will take another look.

I'll investigate that new feed disappearing issue, too.

I'm next scheduled to work on NewsBlur on the week starting October 23rd. Can these wait till then, or are you keen to get it out? I can make some time sooner if needed.

Xcode Cloud for the regular (non-Alpha) TestFlight is linked to the Master branch. We could change that to use the Dejal branch if you wish; that might make sense, since Master should be reserved for general releases.

**Comment by samuelclay** [comment]
I think the dejal branch, or whichever branch is being actively worked on, should be hooked up to Xcode cloud for the NB Alpha build.

No need to rush, but a user-visible issue like this would be good to fix if you find the time.

**Comment by Dejal** [comment]
The Dejal branch has the release-ready features and fixes, so is actively being worked on. And the Catalyst branch has future features (Mac support currently), so is also actively being worked on.

It'd probably make sense to hook the Dejal branch to the regular NewsBlur TestFlight, and leave the Catalyst branch on the NB Alpha TestFlight.

(If you wanted to keep the regular TestFlight as a manual build, maybe we need a NB Beta for the Dejal branch's TestFlight, but that seems unnecessary.)

**Comment by samuelclay** [comment]
> It'd probably make sense to hook the Dejal branch to the regular NewsBlur TestFlight, and leave the Catalyst branch on the NB Alpha TestFlight.

Great, let's do that!

**Comment by Dejal** [comment]
Okay. I've created an Xcode Cloud workflow named "NewsBlur for iOS" that builds when the Dejal branch changes. I've set it to automatically do an internal and external TestFlight. 

I haven't run the workflow yet, since I haven't had time to look into that new feed disappearing issue yet. Let me know if you want me to run it in the meantime for the previous widget tweak. And confirm that you're okay with it automatically doing a TestFlight for everyone.

**Comment by samuelclay** [comment]
Yes I'm ok with it being automatic for everybody. And let's wait to run it until you fix the disappearing feed issue.

**Comment by samuelclay** [comment]
Looks like there's a few reports of a crashing TestFlight install: https://forum.newsblur.com/t/test-flight-install-crashing-and-marking-as-read/10273

**Comment by Dejal** [comment]
Okay, I'll take a look on Monday, once the crash reports make it through Apple's queue.

**Comment by aronatkins** [comment]
Following up to my "feeds appear empty" observation, I have also seen that feeds sometimes continue to enumerate stories that have been read. The website stops showing the stories, but they still appear (though styled-as-read) in the iOS app. This one happens less often and is difficult to capture in a screen recording. If there is something off with how feed state is maintained, I could see that this has a similar root cause, however.

My feeds are configured to show unread-only.

**Comment by Dejal** [comment]
Done a bunch of crash fixes. New TestFlight on its way.

**Comment by samuelclay** [comment]
Why is the build number 88? We launched build 147 a few weeks ago. Am I looking at the wrong upload?
<img width="827" alt="Screenshot 2023-10-24 at 8 39 09 AM" src="https://github.com/samuelclay/NewsBlur/assets/44229/aa8f5520-52ba-423e-a4f5-bb85a55f0c20">

**Comment by Dejal** [comment]
Xcode Cloud has its own build number, starting from 1. But it's possible to update that in the Xcode Cloud settings, so I've changed that; now building 151.

**Comment by aronatkins** [comment]
Update: I have not seen the feed caching problems with build 151 that I reported previously. The comments here do not make it clear if they were explicitly fixed or perhaps addressed as a side-effect with some of the "crash fixes" that were noted.
